generation/WinSDK/enums.json

```diff
@@ -34707,6 +34707,143 @@
       }
     ]
   },
+  {
+    "type": "uint",
+    "name": "DXGI_PRESENT",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_PRESENT_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGISwapChain",
+        "method": "Present",
+        "parameter": "Flags"
+      },
+      {
+        "interface": "IDXGISwapChain1",
+        "method": "Present1",
+        "parameter": "PresentFlags"
+      },
+      {
+        "interface": "IDXGIDecodeSwapChain",
+        "method": "PresentBuffer",
+        "parameter": "Flags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_ENUM_MODES",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_ENUM_MODES_",
+      "header": "dxgi.h|dxgi1_2.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGIOutput",
+        "method": "GetDisplayModeList",
+        "parameter": "Flags"
+      },
+      {
+        "interface": "IDXGIOutput1",
+        "method": "GetDisplayModeList1",
+        "parameter": "Flags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_MWA",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_MWA_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGIFactory",
+        "method": "MakeWindowAssociation",
+        "parameter": "Flags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_MAP",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_MAP_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGISurface",
+        "method": "Map",
+        "parameter": "MapFlags"
+      }
+    ]
+  },
+  {
+    "type": "uint",
+    "name": "DXGI_RESOURCE_PRIORITY",
+    "flags": true,
+    "autoPopulate": {
+      "filter": "DXGI_RESOURCE_PRIORITY_",
+      "header": "dxgi.h"
+    },
+    "members": [],
+    "uses": [
+      {
+        "interface": "IDXGIResource",
+        "method": "SetEvictionPriority",
+        "parameter": "EvictionPriority"
+      },
+      {
+        "interface": "IDXGIResource",
+        "method": "GetEvictionPriority",
+        "parameter": "pEvictionPriority"
```

Comment:
Looking at windows-rs codegen, this generates correctly 🎉. The diff however changes the ABI:

```
Windows.Win32.Graphics.Dxgi.IDXGIResource.GetEvictionPriority : pEvictionPriority...UInt32* => DXGI_RESOURCE_PRIORITY*
```

That isn't desired. Same for all the other non-pointer type changes introduced by the last commit.

Reply:
If that change isn't desired, you'll have to remove it from `uses` in the json above.

Reply:
@riverar I think what I meant to say is in-line with what was discussed above: this should have an `[AssociatedEnum("DXGI_RESOURCE_PRIORITY")]` attribute?

Is that something I should set up via `emitter.settings.rsp` instead of the `uses` list? What about all the other type changes shown in `ChangesSinceLastRelease`?

The same comment/thread is open at https://github.com/microsoft/win32metadata/pull/1757#discussion_r1455979179.

Reply:
Set AssociatedEnum in emitter settings if you don't want to remap the type. Plenty of examples there.

Reply:
As long as the underlying type of the enum matches the type where it's applied, you're not changing the ABI. The build will fail if you change the ABI with an enum.

Reply:
@mikebattista that seems very strange to me. (It was told that the underlying type of an `enum` is `int`](https://github.com/microsoft/win32metadata/pull/1757#discussion_r1426155306) yet as shown in https://github.com/microsoft/win32metadata/pull/1757#discussion_r1455979179 all these types were changed from `UInt32` to an `int`eger based `enum` (which is signed), yet the build did not fail.

It might be useful to automatically have `"uses"` emit a type change _or_ `[AssociatedEnum]` depending on the original type, if such a thing is even possible?

---

Anyway, I _think_ `[AssociatedEnum]` is then the right way to go for all of these?

Reply:
Scratch that, the reply above is completely wrong (talking about `enum` types being `int`, when this PR concerns itself with `#define`s scraped _into_ enums).

What I see in ILSpy for these `enum`s is that they have a `: uint` type, and the `#define`s are all defined with `UL` constant suffix, further clarifying that their type is `UINT`. Accordingly, these `enum`s are the same type as the original `UINT` fields/parameters and can be safely replaced without changing ABI, hence the build succeeding and hence no `AssociatedEnum` being needed.